agent/connect_ca_endpoint.go

```diff
@@ -37,18 +38,27 @@ func (s *HTTPHandlers) ConnectCARoots(resp http.ResponseWriter, req *http.Reques
 
 	// defined in RFC 8555 and registered with the IANA
 	resp.Header().Set("Content-Type", "application/pem-certificate-chain")
-	for _, root := range reply.Roots {
-		if _, err := resp.Write([]byte(root.RootCert)); err != nil {
-			return nil, err
+
+	err := writeCA(resp, reply.Roots)
+	if err != nil {
+		return nil, err
+	}
+
+	return nil, nil
```

Comment:
You could simplify this to just this if you wanted:

```suggestion
	return nil, err
```